sort_json.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env python
+"""
+This is a modification of the pretty_format_json hook downloaded 2022-06-21
+from https://github.com/pre-commit/pre-commit-hooks/blob/b0f06dc9f2260909f3423243de180edfc823ec5a/pre_commit_hooks/pretty_format_json.py
+
+Copyright (c) 2014 pre-commit dev team: Anthony Sottile, Ken Struys
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+from __future__ import annotations
+
+import argparse
+import json
+import sys
+from difflib import unified_diff
+from typing import Mapping
+from typing import Sequence
+
+
+def _get_pretty_format(
+        contents: str,
+        indent: str,
+        ensure_ascii: bool = True,
+        sort_keys: bool = True,
+        top_keys: Sequence[str] = (),
+        sorted_keys: Sequence[str] = (),
+) -> str:
+    def pairs_first(pairs: Sequence[tuple[str, str]]) -> Mapping[str, str]:
+        # Start of the modification
+        new = []
+        for name, value in pairs:
+            if name in sorted_keys:
+                value = list(set(value))
+                value.sort(key=lambda item: item.lower())
+            new.append((name, value))
+        pairs = new
+        # End of the modification
```

Comment:
Is this change perhaps something we can push upstream, if not it feels like a large script to keep around if all we want is this:

```python
from argparse import ArgumentParser
import json
from pathlib import Path
from typing import Optional, Sequence

def main(argv: Optional[Sequence[str]] = None) -> int:
    parser = ArgumentParser()
    parser.add_argument("--sort-keys", type=lambda v: v.split(","), default=[], help="Keys to sort")
    parser.add_argument("filenames", nargs="*", type=Path, help="Filenames to fix")
    args = parser.parse_args(argv)

    def object_pairs_hook(pairs):
        return {
            name: sorted(set(value), key=str.lower) if name in args.sort_keys else value
            for name, value in pairs
        }

    status = 0
    for path in args.filenames:
        original = path.read_text()
        try:
            content = json.loads(original, object_pairs_hook=object_pairs_hook)
        except ValueError:
            status = 2  # invalid json format
        else:
            pretty = f"{json.dumps(content, indent=2)}\n"  # include trailing newline
            path.write_text(pretty)
            status |= original != pretty  # record whether modifications were made
    return status

if __name__ == '__main__':
    raise SystemExit(main())
```

Reply:
I'm fine just using what you posted, not thinking my updates are useful for upstream but I could be wrong.

Reply:
Fixed in https://github.com/conda/infra/pull/565/commits/4e13e4d1d99eadbb1c1fd4c865a1fb6636f6dfbd.